src/characters/pre-recipient.ts

```diff
@@ -12,23 +12,22 @@ import { Condition, ConditionContext } from '../conditions';
 import { Keyring } from '../keyring';
 import { PolicyMessageKit } from '../kits/message';
 import { RetrievalResult } from '../kits/retrieval';
+import { PorterClient } from '../porter';
 import { base64ToU8Receiver, bytesEquals, toJSON, zip } from '../utils';
 
-import { Porter } from './porter';
-
-export type PreTDecDecrypterJSON = {
+export type PreDecrypterJSON = {
   porterUri: string;
   policyEncryptingKeyBytes: Uint8Array;
   encryptedTreasureMapBytes: Uint8Array;
   publisherVerifyingKeyBytes: Uint8Array;
   bobSecretKeyBytes: Uint8Array;
```

Comment:
If this is no longer PreTDecDecrypter, and just PreDecrypter, then bob's secret keys should no longer be needed, right?

Reply:
How does the `PreDecrypter` decrypt without a secret key?

Reply:
For PRE-tDEC Bob's secret key bytes were publicly known anyway so I thought that's why they were passed in here i.e. the Universal Bob concept. 

However, for just PRE, Bob's secret bytes should continue to be a secret to everyone except for Bob. So a few things stand out:
1. Should a `PREDecrypter` be handling Bob's secret bytes at all - that seems like a security hole. More philosophically, what is a "decrypter" in the context of PRE? For example, does a PRE decrypter return re-encrypted bytes (encrypted for Bob) that Bob can then decrypt himself instead? Basically it gets the data re-encrypted and is separated from decryption. It's kind of like the line between Porter and Bob or in Python the line between `PRERetrievalClient` vs `Bob`. In which case, it's not a "decrypter" per se.

2. If `PREDecrypter`s are persistable they should not be persisting bob's secret bytes. Bob should maintain control of his own secret key, which brings me back to point 1) above.

I might need to go back and look at the existing code, maybe I'm just missing something - I think this is where the API becomes very tricky 😅

Reply:
I thing I have more clarity now on where to go from there: It feels to me like `PREDecrypter` is no longer a satisfactory abstraction for decryption, and that job should be delegated to `Bob`. And in that case, maybe we should also revisit the shape and function of `PreStrategy`.

These item look like a big job, and I'd rather sketch a refactoring in a separate issue (#166), discuss it, and then fix it in another PR.